### PR TITLE
Improve water and reflective surfaces in Star Fox Adventures

### PR DESCRIPTION
**Pull request by beholdnec**
This PR improves the look of water, and implements the reflective floors in Krazoa Palace (which looked rather bizarre in the game, too).

### CODE REVIEW

File: src/StarFoxAdventures/shaders.ts [hunk 1]
```diff
@@ -638,6 +699,23 @@ export function buildMaterialFromShader(device: GfxDevice, shader: Shader, texCo
 
     return {
         material: mb.finish(),
-        textures
+        textures,
+        setupMaterialParams: (params: MaterialParams, viewerInput: ViewerRenderInput, modelViewMtx: mat4) => {
+            for (let i = 0; i < 10; i++) {
+                params.u_TexMtx[i] = texMtx[i](viewerInput, modelViewMtx);
```
Comment (magcius) [inline src/StarFoxAdventures/shaders.ts]: Why not `texMtx[i](params.u_TexMtx[i], viewerInput, modelMatrix)` and prevent the extra copy?

File: src/StarFoxAdventures/shaders.ts [hunk 2]
```diff
@@ -638,6 +699,23 @@ export function buildMaterialFromShader(device: GfxDevice, shader: Shader, texCo
...
+            }
+            
+            for (let i = 0; i < 3; i++) {
+                if (indTexMtx[i] !== null) {
+                    params.u_IndTexMtx[i] = indTexMtx[i]!;
```
Comment (magcius) [inline src/StarFoxAdventures/shaders.ts]: I would prefer `mat4.copy` rather than re-assignment.

File: src/StarFoxAdventures/shaders.ts [hunk 3]
```diff
@@ -560,13 +607,20 @@ export function buildMaterialFromShader(device: GfxDevice, shader: Shader, texCo
...
 
-        // TODO: GXSetIndTexMtx
+        indTexMtx[2] = mat4.create();
+        mat4.set(indTexMtx[2],
+            0.0, 0.5, 0.0, 0.0,
+            -0.5, 0.0, 0.0, 0.0,
+            0.0, 0.0, 1.0, 0.0,
+            0.0, 0.0, 0.0, 1.0
+        );
+        mat4.transpose(indTexMtx[2], indTexMtx[2]);
```
Comment (magcius) [inline src/StarFoxAdventures/shaders.ts]: Could you just inline the transpose here? I think the only difference is where you put the negative sign.

**Comment by beholdnec** [comment]
Updated to address feedback.

**Pull request merged**